[6.2.z] Updated manifest deletion to use confirmation dialog (BZ1266827)

abalakh opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1266827
```python
py.test tests/foreman/ui/test_subscription.py -k test_positive_upload_and_delete
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1
collected 2 items

2017-05-25 14:11:33 - conftest - DEBUG - Found WONTFIX in decorated tests ['1110476', '1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-05-25 14:11:33 - conftest - DEBUG - Collected 2 test cases


tests/foreman/ui/test_subscription.py .


============================== 1 tests deselected ==============================
================== 1 passed, 1 deselected in 1001.34 seconds ===================
```
Part of #4733

oshtaier commented:
And where is a test case that covering that functionality? That will test dialog content and that buttons are really working
Or I am missing something?

abalakh commented:
@oshtaier PR is more about updating robottelo/fixing failures rather than adding coverage. The coverage itself is present too - we do have at least `test_positive_upload_and_delete` and `test_positive_delete_manifest` which are attempting to delete manifest. In case confirmation dialog is not there or buttons aren't working, mentioned tests will fail, so i don't think separate test for that specific BZ is necessary, it's basically duplication.
However it may be worth adding additional test to cover `cancel` button. @oshtaier your thoughts?

oshtaier commented:
Yeah, if you don't mind
+
```
A confirmation step will address this confusion by **explaining** to users refresh vs delete and asking if they're sure they want to delete manifest or just go back and refresh it.
```
Maybe it will be good to check message itself anyway

ldjebran commented on robottelo/ui/subscription.py:
```diff
@@ -42,13 +42,17 @@ def upload(self, manifest, repo_url=None):
-            locators['subs.manifest_exists'], timeout)
+        if really:
+            self.click(common_locators['confirm_remove'])
+            timeout = 900 if bz_bug_is_open(1339696) else 300
```
rewrite like:
```
timeout = 300
if bz_bug_is_open(1339696):
    timeout = 900
```

oshtaier replied:
not sure that I understand the difference

abalakh replied:
Easier to delete a workaround later when BZ is closed, i guess. Anyway, that's a low-priority nitpick as for me, but i don't mind :)

ldjebran commented:
this PR has tag test_qe_coverage so it has to test the RFE steps requested, or tag it testfailure and remove test_qe_coverage (as in the current version it has no connection with the RFE bug)

abalakh commented:
@ldjebran by merging this PR we should be able to set qe_test_coverage '+' for mentioned BZ, that's why i put it here. Another test with exact steps from BZ is not needed as we already have such coverage, the only thing missing is this framework change. Anyway i'm going to add another test for 'cancel' button as @oshtaier suggested so this tag is needed here for sure. Re test failure suggestion - i thought we use it for issues only, to be able to quickly find the ones related to failures, don't we?

ldjebran commented:
ACK, pending comment about timeout

abalakh commented:
Comments addressed. @oshtaier regarding verification of message itself - currently there's no message at all, so even if we decide to verify it, it can't be done right now (although it's debatable, i'm not sure it's a good idea to rely on localized string which may change in future or even now if you select different language). @oshtaier any suggestions? Personally i'd like to merge this PR regardless of our decision about message - it contains important fix, tests are failing without it.

abalakh commented:
```python
py.test tests/foreman/ui/test_subscription.py -k 'test_positive_upload_and_delete or test_negative_delete'
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.15.0, services-1.1.14, cov-2.3.1


Added test and addressed comments
collected 3 items
2017-05-30 13:30:19 - conftest - DEBUG - Found WONTFIX in decorated tests ['1110476', '1156555', '1269196', '1402826', '1245334', '1221971', '1217635', '1226425', '1199150', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-05-30 13:30:19 - conftest - DEBUG - Collected 3 test cases


tests/foreman/ui/test_subscription.py ..

============================== 1 tests deselected ==============================
=================== 2 passed, 1 deselected in 208.54 seconds ===================
```

oshtaier commented:
ACK
Message is important because of corresponding accent on that in the defect itself, but definitely priority of that is not so high
But, if `currently there's no message at all` is true then defect is not fixed, because it is a wrong behavior and my initial idea was to catch cases like that
